inc/dbmysql.class.php

```diff
@@ -209,7 +209,8 @@ function connect($choice = null) {
          }
 
          // force mysqlnd to return int and float types correctly (not as strings)
-         if (GLPI_FORCE_NATIVE_SQL_TYPES) {
+         // we still check if constant is defined to prevent warnings during requirements checks
+         if (defined('MYSQLI_OPT_INT_AND_FLOAT_NATIVE')) {
```

Comment:
i think we need to hard break here if constant is not defined

Reply:
This check is only here to prevent an error in this case:
![image](https://user-images.githubusercontent.com/33253653/87323308-e8048800-c52e-11ea-8c32-e24e55801c5d.png)

Reply:
Do you want me to add an error log ?

Reply:
I guess the point was to ensure user cannot run glpi at all with the wrong one; a log would not be enough in that case.

Reply:
I added a commit to block usage of GLPI without the good driver, and make requirements checks working without warning.